packages/swayze/lib/src/core/controller/selection/selection_controller.dart

```diff
@@ -73,11 +86,20 @@ class SwayzeSelectionController extends Listenable implements ControllerBase {
   /// version.
   /// If the returned state is the same as before, no update is triggered.
   void updateUserSelections(
-    UserSelectionState Function(UserSelectionState previousState) stateUpdate,
+    _StateChanger<UserSelectionState> stateUpdate,
   ) {
     userSelections.value = stateUpdate(userSelections.value);
   }
 
+  /// Update the current [fillSelectionState]. The callback [stateUpdate]
+  /// receives the actual version of the state and should return the updated
+  /// version.
+  /// If the returned state is the same as before, no update is triggered.
+  void updateFillSelections(
+    _StateChanger<FillSelectionState> stateUpdate,
+  ) =>
+      fillSelection.value = stateUpdate(fillSelection.value);
```

Comment:
still waking up, so I could be misinterpreting something here, but the doc says "If the returned state is the same as before, no update is triggered.", but we have no condition based on the returned state in the implementation

Reply:
Good catch. I added the doc to remind myself, but this code changed and I forgot to check. Adding.